1-ECommerce/src/components/Menu.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from "react";
+import type { Option } from "../interfaces";
+import MenuOption from "./MenuOption";
+import Sidebar from "./Sidebar";
+
+
+export default function Menu(): JSX.Element {
+    const [isMenuVisible, setIsMenuVisible] = useState<boolean>(false)
+    const [selectedOption, setSelectedOption] = useState<number>(0)
+
+    const handleMenuClick = (): void => {
+        setIsMenuVisible(true)
+    }
+
+    const handleCloseMenu = (): void => {
+        setSelectedOption(0)
+        setIsMenuVisible(false)
+    }
+
+    const handleMenuOptionClick = (optionId: number): void => {
+        const newOptionValue: number = (selectedOption === optionId) ? 0 : optionId;
+        setSelectedOption(newOptionValue)
+    }
+
+    const options: Option[] = [
+        { id: 1, name: 'New In'},
+        { id: 2, name: 'Clothing'},
+        { id: 3, name: 'Footwear'},
+        { id: 4, name: 'Accesories'},
+        { id: 5, name: 'SALE'}
+    ]
+
+    const subMenuOptions: Option[] = [
+        { id: 11, name: 'New In' },
+        { id: 12, name: 'See All'},
+        { id: 13, name: 'Coats'},
+        { id: 14, name: 'Beach clothes'},
+        { id: 15, name: 'Sweaters & hoodies'},
+        { id: 16, name: 'Shirts'},
+        { id: 17, name: 'Jeans & pants'},
+        { id: 18, name: 'T-shirts'},
+        { id: 19, name: 'Shorts'},
+        { id: 20, name: 'Underwear'},
+        { id: 21, name: 'SALE'}
+    ]
+
+    const menuOptions: JSX.Element[] = options
+        .map(option => {
+            return (
+                <MenuOption
+                    key={option.id}
+                    active={selectedOption === option.id}
+                    option={option}
+                    onOptionClick={() => handleMenuOptionClick(option.id)}
+                />
+            )
+    })
+
+    const hardcodedSubMenuOptions: JSX.Element[] = subMenuOptions
+        .map(option => {
+            return (
+                <MenuOption
+                    key={option.id}
+                    active={false}
+                    option={option}
+                    onOptionClick={() => alert(`suboption "${option.name}" clicked`)}
+                />
+            )
+    })
```

Comment:
When we just return a component, we can use this shorter syntax:

```suggestion
        .map(option => (
                <MenuOption
                    key={option.id}
                    active={false}
                    option={option}
                    onOptionClick={() => alert(`suboption "${option.name}" clicked`)}
                />
            ))
```